### ISSUE TITLE
Split tunneling blocking connection for specific app (Java)

### ISSUE THREAD

**Issue by Gwiggle** [mullvad/mullvadvpn-app#3899]
Operating system: Windows 11

App version: 2022.4

I'm trying to push a game through split tunneling so I don't have a higher latency and it seems to be completely blocking the connection. Leaving split tunneling off and the game connects through the vpn fine and having the vpn off obviously works but trying to use split tunneling just refuses the connection. It did use to work but I'm not sure what could have changed to start blocking it. Any ideas?

The game does spit out an error in their log

```
java.net.SocketException: Permission denied: connect
	at java.base/java.net.PlainSocketImpl.connect0(Native Method)
	at java.base/java.net.PlainSocketImpl.socketConnect(Unknown Source)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(Unknown Source)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(Unknown Source)
	at java.base/java.net.AbstractPlainSocketImpl.connect(Unknown Source)
	at java.base/java.net.SocksSocketImpl.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at java.base/sun.net.NetworkClient.doConnect(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.http.HttpClient.openServer(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.<init>(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsClient.New(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.plainConnect(Unknown Source)
	at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(Unknown Source)
	at net.runelite.launcher.Launcher.getBootstrap(Launcher.java:452)
	at net.runelite.launcher.Launcher.main(Launcher.java:262)
```

Along with the game .exe I've also tried passing java through the tunnel as well with no success

**Comment by faern** [comment]
> It did use to work but I'm not sure what could have changed to start blocking it. Any ideas?

When did it work? did it work in a previous app version and stopped working with a specific upgrade? If so, what version broke this splitting?

**Comment by Gwiggle** [comment]
I believe I updated the launcher for it to break. It had always worked
before that. I downgraded back to the previous version of the launcher I
was using and it was still not working. It's a Java based application which
I've found online vpns don't like sometimes? The launcher itself gives me
an error of 'java.net.socketexception: permission denied: connect'.

It's a Java based game that I'd like to have split tunneling for to reduce
latency.
